eixampledb-api/src/main/java/com/eixampledb/api/MainController.java

```diff
@@ -27,54 +25,36 @@
         } else {
             return ResponseEntity.notFound().build();
         }
     }
 
     @RequestMapping(path = "/{key}", method = RequestMethod.POST)
-    public ResponseEntity set(@PathVariable("key") String key, @RequestBody String value, @RequestHeader Map<String,String> header) {
-        String type = header.get("type");
-        int t;
-        if (type == null){
-            type = "";
-        }
-        if (type.equals("NUM")){
-            t = core.NUM_TYPE;
-        }else if (type.equals("STR")){
-            t = core.STRING;
-        }
-        else{
-            t = core.STRING;
-        }
-        eixampledb.set(new SetRequest(key, value, t));
+    public ResponseEntity set(@PathVariable("key") String key,
+                              @RequestBody String value,
+                              @RequestHeader(name = "type", defaultValue = "STR", required = false) ValueType valueType) {
+        eixampledb.set(new SetRequest(key, value, valueType));
         return ResponseEntity.ok().build();
     }
 
     @RequestMapping(path = "/{key}", method = RequestMethod.PUT)
-    public ResponseEntity operation(@PathVariable("key") String key, @RequestHeader("op") String operation) {
-
-        GetResponse getResponse = eixampledb.get(new GetRequest(key));
-        String value = getResponse.getEntry().get().getValue(); //Value of the key
-        int type = getResponse.getEntry().get().getType(); // Type of the key
-
-        if (! getResponse.isSuccess()) { //If the key doesn't exist return error
-            return ResponseEntity.notFound().build();
+    public ResponseEntity operation(@PathVariable("key") String key,
+                                    @RequestHeader(value = "op") Operation operation) {
+        Response<?> response;
+        if (operation.isDecrement()) {
+            response = eixampledb.decr(new DecrRequest(key));
+        } else if (operation.isIncrement()) {
+            response = eixampledb.incr(new IncrRequest(key));
+        } else {
+            throw new UnsupportedOperationException("Unsupported operaton " + operation.name());
         }
-
-        if(type == core.NUM_TYPE) { //If it's a number
-            if (operation.equals("INCR")) { //If op is INCR, increase the value caring if it's int/long or float/double
-                value = core.operation_increment(key, value);
-            } else if (operation.equals("DECR")) {
-                value = core.operation_decrement(key, value);
-            }
-            SetResponse setResponse = eixampledb.set(new SetRequest(key, value, core.NUM_TYPE));
-        }else{
-            SetResponse setResponse = eixampledb.set(new SetRequest(key, value, type));
+        if (response.isSuccess()) {
+            return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+        } else {
+            return ResponseEntity.notFound().build();
         }
-
-        return ResponseEntity.ok().build();
```

Comment:
OK responses with empty body should return status code 204 to tell that it is ok (2xx family) and it is empty body.